packages/amplify-cli/src/extensions/amplify-helpers/envResourceParams.ts

```diff
@@ -59,23 +61,49 @@ function removeObjectRecursively(obj, keys) {
 }
 
 export function saveEnvResourceParameters(context: $TSContext, category: string, resource: string, parameters: $TSObject) {
+  if (!parameters) return;
   const teamProviderInfo = getApplicableTeamProviderInfo(context);
   const currentEnv = getCurrentEnvName(context);
   const resources = getOrCreateSubObject(teamProviderInfo, [currentEnv, CATEGORIES, category]);
-
-  resources[resource] = _.assign(resources[resource], parameters);
+  const { hostedUIProviderCreds, ...otherParameters } = parameters;
+  resources[resource] = _.assign(resources[resource], otherParameters);
 
   if (!isMigrationContext(context)) {
     stateManager.setTeamProviderInfo(undefined, teamProviderInfo);
+    // write hostedUIProviderCreds to deploymentSecrets
+    if (hostedUIProviderCreds) {
+      const deploymentSecrets = stateManager.getDeploymentSecrets();
+      const appId = getAmplifyAppId();
+      const newDeploymentSecrets = getOrCreateSubObject(deploymentSecrets, [appId, currentEnv, category, resource]);
+      newDeploymentSecrets[hostedUIProviderCredsField] = hostedUIProviderCreds;
+      stateManager.setDeploymentSecrets(deploymentSecrets);
+    }
   }
 }
 
 export function loadEnvResourceParameters(context: $TSContext, category: string, resource: string) {
-  const teamProviderInfo = getApplicableTeamProviderInfo(context);
+  const envParameters = {
+    ...loadEnvResourceParametersFromDeploymentSecrets(context, category, resource),
+    ...loadEnvResourceParametersFromTeamprovider(context, category, resource),
```

Comment:
Would there ever be a case where a value in team provider would overwrite a value from deployment secrets when we didn't want to overwrite?

Reply:
No not for new projects and migrated projects